homeassistant/components/binary_sensor/rest.py

```diff
@@ -65,11 +65,6 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         auth = None
 
     rest = RestData(method, resource, auth, headers, payload, verify_ssl)
-    rest.update()
-
-    if rest.data is None:
-        _LOGGER.error("Unable to fetch REST data from %s", resource)
-        return False
```

Comment:
We should `raise PlatformNotReady`. Keep in mind that the platform can't recover if the credentials or header entries are wrong. Then the users will end-up with non-functional platforms in their setup.

Reply:
Thanks for the review - I see what you mean.
I am hoping to find a way to distinguish between recoverable and unrecoverable errors. Once the sensor is correctly set up (correct URL, correct username/password, etc.), it would be nice if the code could handle intermittent errors. At present, if a third-party service used in a rest binary sensor is unavailable just at the time when HA is restarted, that binary sensor is not created. 
BTW: A rest sensor is still created in the same situation. And a rest switch is only not created if the HTTP response code is >= 400 or if specific exceptions occur; this may be a role model for sensor and binary sensor?
I'll look into `PlatformNotReady`.

Reply:
Alright, I wasn't aware that when raising a `PlatformNotReady` HA automatically retries after 30 seconds. That definitely helps with intermittent errors during the setup of the sensors.